content/blog/eng/from-0-to-infinite-the-final-keyword.md

```diff
@@ -0,0 +1,338 @@
+---
+authors: ["bruce"]
+date: "2018-12-04"
+draft: true
+share: false
+categories: [English, php, oop]
+title: "From zero to infinite: the Final keyword"
+type: "post"
+languageCode: "en-EN"
+toc: true
+---
+
+# Google from zero to infinite
+
+Sometimes I do a full immersion on topics of my interest, generally related to programming, topics on testing, good design, etc. 
+I apply a technique invented by me named: **"Google from zero to infinite"**.
+
+Basically I use one or more keywords in Google, then I literally follow all the links of all the pages. It is a very long and laborious activity. But most of the time I find very interesting documents, otherwise difficult to reach. This time I decided to use the keyword final of PHP.
+
+# The concept behind final keyword
+ 
+The keyword final was introduced in version 5 of php. 
+
+The manual says: 
+
+> Final Keyword. PHP 5 introduces the final keyword, which prevents child classes from overriding a method by prefixing the definition with final. 
+> If the class itself is being defined final then it cannot be extended. Note: Properties cannot be declared final, only classes and methods may be declared as final.
+
+Java has the same keyword, c# too but it is called "sealed" instead of "final", but more or less have the same behavior: to prevent a class from being inherited (in our case we will see only this type of use, for example languages like Java support further behavior but in different contexts).
+
+First of all we will try to understand what it is for, then we will try to do some additional reasoning, and finally we will talk about of comments on the net.
+
+The concept that expresses the keyword final is that a class can not be extended, the decision must be taken by the programmer in the design stage utilizing just the final keyword. 
```

Comment:
I would change the sentence as follows:
The concept expressed by the final keyword is that a class cannot be extended. The programmer must take this decision at the design stage by using this keyword.

_Note:_ "Cannot" is a much more popular spelling than "can not". I would use "cannot".